modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -3324,6 +3418,9 @@ protected void removeAll0(final Collection<? extends K> keys) throws IgniteCheck
         if (statsEnabled)
             fut.listen(new UpdateRemoveTimeStatClosure<>(metrics0(), start));
 
+        if (profilingEnabled)
+            fut.listen(new ProfileClosure<>(CacheOperationType.REMOVE_ALL, start));
```

Comment:
We should have separate class for each operations so we can eliminate `operation` variable from `ProfileClosure`.

Reply:
I think we should push down methods `startTime` and `duration` from `GridCacheFuture` to `IgniteInternalFuture` therefore we can use the single non-static inner class instance to gather statistics both for cache statistics and performance statistics.

Reply:
> We should have separate class for each operations so we can eliminate `operation` variable from `ProfileClosure`.

I have used lambda expressions.